### PR TITLE
fix all items changed handler

### PR DESCRIPTION
**Pull request by maggu2810**
Implement the all items changed handler of the item registry listener
used in the thing link manager.

Bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=457808
Signed-off-by: Markus Rathgeb <maggu2810@gmail.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingLinkManager.java [hunk 1]
```diff
@@ -82,8 +83,56 @@ public void added(Item element) {
...
+                    }
+                }
+            }
+
+            /*
+             * After all old items are removed, we could call the added handler for all items that are still available.
+             */
+            final Collection<Item> allCurrentItems = itemRegistry.getAll();
+            for (final Item item : allCurrentItems) {
+                logger.debug("add item {}", item.getName());
```
Comment (dnobel) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingLinkManager.java]: Could you please remove the log message here?

**Comment by dnobel** [comment]
Thanks for the PR! The implementation looks really good. Could you please just remove the log message before I will merge it? Thanks!

**Comment by dnobel** [comment]
I forgot: Please add yourself also as an author to the class with a short comment what you have added.

**Comment by maggu2810** [comment]
Done.

**Pull request merged**